### PR TITLE
feat(#939): Upgrade `eo` up to the `0.49.2` version

### PR DESCRIPTION
**Pull request by volodya-lombrozo**
Upgrade `eo` version up to `0.49.2`. Additionally, in this PR I:
1) removed `abstract` objects.
2) removed redundant aliases https://github.com/objectionary/lints/issues/127#issuecomment-2531480301 
3) removed name usage from most of the objects (where it was possible)
4) Added `objectionary/lints` dependency to check generated `XMIR` https://github.com/objectionary/jeo-maven-plugin/issues/938
5) Upgrade `jcabi-xml` to the `0.32.0` version

Related to #939.

### CODE REVIEW

**Comment by volodya-lombrozo** [comment]
Blocked by https://github.com/objectionary/eo/issues/3547

**Comment by volodya-lombrozo** [comment]
Blocked by https://github.com/objectionary/lints/issues/127

File: src/main/java/org/eolang/jeo/representation/XmirRepresentation.java [hunk 1]
```diff
@@ -150,7 +150,7 @@ public String name() {
     public Bytecode toBytecode() {
         final Node xmir = this.xml.value();
         try {
-            new StrictXML(new XMLDocument(new XMLDocument(xmir).toString())).validate();
+            new StrictXmir(new XMLDocument(new XMLDocument(xmir).toString())).inner();
             return new XmlProgram(xmir).bytecode().bytecode();
```
Comment (maxonfjvipon) [inline src/main/java/org/eolang/jeo/representation/XmirRepresentation.java:154]: @volodya-lombrozo you can do here just
```java
new XmlProgram(new StrictXmir(new XMLDocument(xmir)).inner()).bytecode().bytecode()
```

Comment (volodya-lombrozo) [reply src/main/java/org/eolang/jeo/representation/XmirRepresentation.java:154]: @maxonfjvipon I can't remove extra `new XMLDocument`. It's one more bug in `jcabi-xml`. You can try to remove it and you will see that unit tests fail.

**Comment by volodya-lombrozo** [comment]
@rultor merge

**Comment by 0crat** [comment]
@volodya-lombrozo Thanks for the contribution! You've earned +4 points for this: +4 as a basis; -8 for too many hits-of-code (573 >= 200); -16 for the lack of code review; +24 to give you at least something. Please, [keep](https://www.yegor256.com/2018/03/06/speed-vs-quality.html) them coming. Your running balance is [+4](https://www.eolang.org/zerocracy/objectionary-vitals.html).

**Pull request merged**